engine/core/src/test/java/org/datacleaner/test/full/scenarios/TransformationAndFilteringInOutputDataStreamsTest.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * DataCleaner (community edition)
+ * Copyright (C) 2014 Neopost - Customer Information Management
+ *
+ * This copyrighted material is made available to anyone wishing to use, modify,
+ * copy, or redistribute it subject to the terms and conditions of the GNU
+ * Lesser General Public License, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.test.full.scenarios;
+
+import java.util.List;
+
+import org.datacleaner.api.Configured;
+import org.datacleaner.api.Filter;
+import org.datacleaner.api.InputColumn;
+import org.datacleaner.api.InputRow;
+import org.datacleaner.configuration.DataCleanerConfiguration;
+import org.datacleaner.configuration.DataCleanerConfigurationImpl;
+import org.datacleaner.connection.Datastore;
+import org.datacleaner.data.MutableInputColumn;
+import org.datacleaner.job.AnalysisJob;
+import org.datacleaner.job.builder.AnalysisJobBuilder;
+import org.datacleaner.job.builder.AnalyzerComponentBuilder;
+import org.datacleaner.job.builder.FilterComponentBuilder;
+import org.datacleaner.job.builder.TransformerComponentBuilder;
+import org.datacleaner.job.runner.AnalysisResultFuture;
+import org.datacleaner.job.runner.AnalysisRunnerImpl;
+import org.datacleaner.test.MockAnalyzer;
+import org.datacleaner.test.MockOutputDataStreamAnalyzer;
+import org.datacleaner.test.MockTransformer;
+import org.datacleaner.test.TestHelper;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TransformationAndFilteringInOutputDataStreamsTest {
+
+    public static enum BespokeCategory {
+        VALID, INVALID;
+    }
+
+    public static class BespokeNotNullFilter implements Filter<BespokeCategory> {
+
+        @Configured
+        InputColumn<?> inputColumn;
+
+        @Override
+        public BespokeCategory categorize(InputRow inputRow) {
+            Assert.assertTrue(inputRow + " does not contain " + inputColumn, inputRow.containsInputColumn(inputColumn));
```

Comment:
Hehe the message is indeed a error message if the assertion fails. So I think it's fine.